cmd/profile/add.go

```diff
@@ -0,0 +1,216 @@
+			if !validProfileName.MatchString(args[0]) {
+				return fmt.Errorf("%q is not a valid profile name", args[0])
+			}
+			return nil
+		},
+		RunE: func(c *cobra.Command, args []string) error {
+			return addRun(c, args, opts)
+		},
+	}
+}
+
+// defaultProfileName is the profile name if already have a config populated
+// before we run the command
+const defaultProfileName string = "default"
+
+func addRun(cmd *cobra.Command, args []string, opts *commandOpts) error {
+	if !profiles.FileExists() {
+		profileFile, err := os.Create(profiles.FilePath())
+		if err != nil {
+			return fmt.Errorf("unable to create profiles directory %w", err)
+		}
+		defer profileFile.Close()
+		if _, err := profileFile.WriteString("{}"); err != nil {
+			return err
+		}
+	}
+	if !profiles.DirectoryExists() {
+		if err := profiles.CreateDirectories(); err != nil {
+			return fmt.Errorf("could not create profile directory %w", err)
+		}
+	}
+
+	p, err := profiles.Read()
+	if err != nil {
+		return err
+	}
+
+	if len(p.List) == 0 {
+		// if profile.List is empty and we have a existing config.json in %SDPCTL_CONFIG_DIR
+		// migrate the existing profile to a profile before adding a new one
+		rootConfig := filepath.Join(filesystem.ConfigDir(), "config.json")
+		if ok, err := util.FileExists(rootConfig); err == nil && ok {
+			// move to profile default if it has a address in the config
+			config, err := readConfig(rootConfig)
+			if err != nil {
+				return err
+			}
+			if h, err := config.GetHost(); len(h) > 0 && err == nil {
+				directory := filepath.Join(profiles.Directories(), defaultProfileName)
+				if err := os.Mkdir(directory, os.ModePerm); err != nil {
+					return fmt.Errorf("could not create new default config profile directory %w", err)
+				}
+				files, err := moveDefaultConfigFiles(filesystem.ConfigDir(), directory)
+				if err != nil {
+					return err
+				}
+				if err := updatePemPath(directory, files); err != nil {
+					fmt.Fprintf(opts.Out, "could not update PEM file path for default config %s\n", err)
+				}
+				p.List = append(p.List, profiles.Profile{
+					Name:      defaultProfileName,
+					Directory: directory,
+				})
+				p.Current = &directory
+			}
+		}
+	}
+	name := args[0]
+	directory := filepath.Join(profiles.Directories(), name)
+	if err := os.Mkdir(directory, os.ModePerm); err != nil {
+		return fmt.Errorf("profile already exists with the name %s", name)
+	}
+
+	p.List = append(p.List, profiles.Profile{
+		Name:      name,
+		Directory: directory,
+	})
+
+	if err := profiles.Write(p); err != nil {
+		return err
+	}
+	fmt.Fprintf(opts.Out, "Created profile %s, run 'sdpctl profile list' to see all available profiles\n", name)
+	fmt.Fprintf(opts.Out, "run 'sdpctl profile set %s' to select the new profile profile\n", name)
+	return nil
+}
+
+// updatePemPath if we have a existing pem file in the default config, update the pem_filepath
+// to respect the new directory tree
+func updatePemPath(targetDir string, files []string) error {
```

Comment:
Is this needed? `pem_filepath` is converted to an absolute path in configure.go#60